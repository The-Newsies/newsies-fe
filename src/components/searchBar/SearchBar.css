input[type=text] {
  margin-top: .5rem;
  border: 2px solid red;
  border-radius: 10px;
  padding: .2rem;
  margin-right: .3rem;
}

::placeholder {
  padding: .2rem;
}

.searchButton {
  height: 1.4rem;
}

.searchButton:hover {
  background: gray;
  color: white;
}