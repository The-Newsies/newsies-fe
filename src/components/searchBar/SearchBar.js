import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './SearchBar.css';

function SearchBar({ handleClick }) {
  const [search, setSearch] = useState('');

  return (
    <div>
      <input type="text" placeholder="search" value={search} onChange={({ target }) => setSearch(target.value)}></input>
      <button onClick={() => handleClick(search)}>Search!</button>
    </div>
  );
}

SearchBar.propTypes = {
  handleClick: PropTypes.func.isRequired
};

export default SearchBar;
