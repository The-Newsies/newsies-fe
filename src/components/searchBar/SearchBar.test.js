import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from './SearchBar';

describe('SearchBar component', () => {
  it('renders SearchBar', () => {
    const wrapper = shallow(<SearchBar text={'hello'} handleOnChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
