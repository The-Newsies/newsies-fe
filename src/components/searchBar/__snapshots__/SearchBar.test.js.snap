// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders SearchBar 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="search all news"
    type="text"
    value=""
  />
  <button
    className="searchButton"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
